cpp/include/Ice/ServerAuthenticationOptions.h

```diff
@@ -0,0 +1,183 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+#ifndef ICE_SERVER_AUTHENTICATION_OPTIONS_H
+#define ICE_SERVER_AUTHENTICATION_OPTIONS_H
+
+#include "SSLConnectionInfo.h"
+
+#include <functional>
+
+#if defined(_WIN32)
+// We need to include windows.h before wincrypt.h.
+// clang-format off
+#    ifndef NOMINMAX
+#        define NOMINMAX
+#    endif
+#    include <windows.h>
+#    include <wincrypt.h>
+// clang-format on
+// SECURITY_WIN32 or SECURITY_KERNEL, must be defined before including security.h indicating who is compiling the code.
+#    ifdef SECURITY_WIN32
+#        undef SECURITY_WIN32
+#    endif
+#    ifdef SECURITY_KERNEL
+#        undef SECURITY_KERNEL
+#    endif
+#    define SECURITY_WIN32 1
+#    include <schannel.h>
+#    include <security.h>
+#    include <sspi.h>
+#    undef SECURITY_WIN32
+#elif defined(__APPLE__)
+#    include <Security/SecureTransport.h>
+#    include <Security/Security.h>
+#else
+#    include <openssl/ssl.h>
+#endif
+
+namespace Ice::SSL
+{
+    /**
+     * The SSL configuration properties for server connections.
+     */
+    struct ServerAuthenticationOptions
+    {
+#if defined(_WIN32)
+        /**
+         * A callback that allows selecting the server credentials based on the server's adapter name.
```

Comment:
I find the term "server's adapter name" unclear.

I think you mean: 
A callback that allows selecting the server credentials based on the name of the object adapter that accepts the connection.